### ISSUE TITLE
[Bug] Unable to find the `dll` when build on Windows in v1 with melos template

### ISSUE THREAD

**Issue by AlexV525** [fzyzcjy/flutter_rust_bridge#1541]
### Describe the bug

1. Create packages with frb and melos.
2. Integrate rust native code in the `library_ffi` package.
3. Add `library_flutter` as a dependency.
4. Build the Flutter app on Windows.
5. Check `build\windows\x64\runner\Debug` and no `library.dll` found.

### Codegen logs with `RUST_LOG=debug` environment variable

```shell
N/A
```

### To Reproduce

1. Clone https://github.com/AstroxNetwork/agent_dart
2. Switch to the `bdk_lite` branch
3. Sync and build the Windows library by running `scripts/build-windows.bat`.
4. Check `build\windows\x64\runner\Debug` to find `agent_dart.dll`.

### OS

Windows

### Version of `flutter_rust_bridge_codegen`

1.82.6

### Flutter info

```shell
[!] Flutter (Channel [user-branch], 3.16.4, on Microsoft Windows [Version 10.0.22621.2861], locale en-US)
    ! Flutter version 3.16.4 on channel [user-branch] at X:\SDK\flutter-master
    • Framework revision 2e9cb0aa71 (3 weeks ago), 2023-12-11 14:35:13 -0700
    • Engine revision 54a7145303
    • Dart version 3.2.3
    • DevTools version 2.28.4
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Windows Version (Installed version of Windows is version 10 or higher)
```

### Version of `ffigen`

9.0.1

**Comment by AlexV525** [comment]
Plugins created by `flutter create --template plugin_ffi --platform=windows` can produce the DLL successfully. I've tried to copy some of its `CMakeLists.txt` but it does not work.

**Comment by fzyzcjy** [comment]
Hi, frb v2 uses cargokit by default to handle the build process, and docs talk about that by default. It seems that you are using custom build script (e.g. https://github.com/AstroxNetwork/agent_dart/blob/61f28f5a1853d2045702326be890491c02c8aaa7/scripts/build-windows.bat#L33), thus just ignore things about building mentioned in FRB doc. e.g. the script seems to use https://github.com/rust-cross/cargo-xwin thus need to consider docs of cargo-xwin.

**Comment by AlexV525** [comment]
1. I'm on the v1 branch, so I'm using the same build mechanism as https://cjycode.com/flutter_rust_bridge/v1/library/platform_setup/windows_and_linux.html suggested. No matter `bat` or `sh`, they are using the same thing.
2. To be clear, `dll`s produces successfully after build, but doesn't seem to be copied to the Flutter script directory.

**Comment by AlexV525** [comment]
I assume the same thing will happen to Linux because they are using a similar configuration.

**Comment by fzyzcjy** [comment]
Hmm, so, does the bug happen after you upgrade v1 to some specific version, or is it always the case?

**Comment by AlexV525** [comment]
> Hmm, so, does the bug happen after you upgrade v1 to some specific version, or is it always the case?

Not quite sure though. I'll try with 1.5x later. I think this started after we applied Melos management.

**Comment by fzyzcjy** [comment]
> I'll try with 1.5x later.
> I think this started after we applied Melos management.

I personally think this may not have much differences, because v1.x is completely unaware of the specific approach you use to glue Rust with Dart. I asked it because I thought, if someone has used FRB and creates an issue today, it must be because of upgrade. So I still suspect it may because the script is somehow broken given the information.

**Comment by AlexV525** [comment]
> I personally think this may not have much differences, because v1.x is completely unaware of the specific approach you use > Then I guess, in addition to directly debug what is going on, another way is to bisect the code of `agent_dart` and see which commit introduces the problem.

Yeah, bisecting. We have a branch (the main branch) that is before melos but uses a pretty new version of FRB. That can be used for tests.

**Comment by AlexV525** [comment]
On the other hand, so you have any template that is using the Melos and includes Windows build? I've searched a few and they either have no Windows support or no Melos integrated.

**Comment by fzyzcjy** [comment]
> do you have any template that is using the Melos and includes Windows build? I've searched a few and they either have no Windows support or no Melos integrated.

FRB v2's default template uses https://github.com/irondash/cargokit. If you are developing a package (instead of a flutter app), you can directly see the blog linked in their README to setup. However, cargokit packages the Rust source code instead of a precompiled binary, so I am not sure whether this suits your needs.

I have not heard of other melos + windows + package template... :/

**Comment by AlexV525** [comment]
Well I basically follow the instruction at here: https://cjycode.com/flutter_rust_bridge/v1/library/melos.html

**Comment by fzyzcjy** [comment]
(For "I have not heard of other melos + windows + package template..." I mean templates excluding the one inside FRB)

**Comment by AlexV525** [comment]
I'm trying to get rid of melos now. :)

**Comment by AlexV525** [comment]
Oh, good news. It did work with our no-melos branch. I'll keep tracking about the difference and provide feedback.

**Comment by AlexV525** [comment]
(https://github.com/AstroxNetwork/agent_dart/commit/15e37b627f6117a2e0bbc8d710fa13bea0f6f4dc)

So there are two things I've misconfigured for the script:
1. When setting the bundled libraries path, its name should be equal to the project name, and the project name should be equal to the package name (in the pubspec.yaml).
2. I forgot the `CMAKE_CURRENT_SOURCE_DIR` in the path since I've changed the extract path.

I think I can improve the template based on the commit. I'll setup a request later.
